contracts/Protected.sol

```diff
@@ -69,19 +69,25 @@ contract Protected {
         bytes32 _id,
         address _to,
         bool _assignable,
-        uint _expiration,
-        uint _uses
+        uint80 _startTime,
+        uint80 _expiration,
+        uint80 _uses
     ) public
     {
         Key memory key = keys[_id][msg.sender];
         require(key.exists && isValidExpiration(key.expiration), "Invalid key");
         require(key.assignable, "Key is not assignable");
+        require(key.startTime == 0 || (_startTime >= key.startTime && _startTime > 0), "Cannot reduce key's start time");
         require(key.expiration == 0 || (_expiration <= key.expiration && _expiration > 0), "Cannot extend key's expiration");
-        require(key.uses == 0 || (_uses <= key.uses && _uses > 0), "Not enough uses avaiable");
+        require(_startTime == 0 || _expiration == 0 || _startTime < _expiration, "Start time must be strictly less than expiration");
         require(isValidExpiration(_expiration), "Expiration must be in the future");
+        require(_startTime == 0 || _startTime >= now, "Start time must be in the future");
```

Comment:
This condition doesn't make sense. Let's say I have a key with startTime of yesterday, now I want to transfer it without start time delay. This makes it impossible.

Reply:
I agree. No reason _startTime should not be in the past.

Reply:
My thinking was that there's no point in having `startTime` yesterday because that time has passed. But yeah, it's unnecessary, and it means you can't use `grantFullKey` in that case.